tempodb/backend/local/local.go

```diff
@@ -104,13 +116,21 @@ func (rw *Backend) CloseAppend(_ context.Context, tracker backend.AppendTracker)
 	return dst.Close()
 }
 
-func (rw *Backend) Delete(_ context.Context, name string, keypath backend.KeyPath, _ bool) error {
+func (rw *Backend) Delete(ctx context.Context, name string, keypath backend.KeyPath, _ bool) error {
+	if err := ctx.Err(); err != nil {
```

Comment:
cc @zalegrala heads up you will want this context logic in the Blocks() method you added to the local backend